libminifi/test/unit/GeneralUtilsTest.cpp

```diff
@@ -32,18 +32,20 @@ TEST_CASE("GeneralUtils::make_unique", "[make_unique]") {
   REQUIRE("test string" == *pstr);
 }
 
-TEST_CASE("GeneralUtils::intdiv_ceil", "[intdiv_ceil]") {
-  REQUIRE(0 == utils::intdiv_ceil(0, 1));
-  REQUIRE(0 == utils::intdiv_ceil(0, 2));
-  REQUIRE(1 == utils::intdiv_ceil(1, 2));
-  REQUIRE(1 == utils::intdiv_ceil(1, 3));
-  REQUIRE(1 == utils::intdiv_ceil(3, 3));
-  REQUIRE(2 == utils::intdiv_ceil(4, 3));
-  REQUIRE(2 == utils::intdiv_ceil(4, 3));
-  REQUIRE(0 == utils::intdiv_ceil(-1, 3));
-  REQUIRE(-1 == utils::intdiv_ceil(-3, 3));
-  REQUIRE(-1 == utils::intdiv_ceil(-4, 3));
-}
+static_assert(0 == utils::intdiv_ceil(0, 1), "");
+static_assert(0 == utils::intdiv_ceil(0, 2), "");
+static_assert(1 == utils::intdiv_ceil(1, 2), "");
+static_assert(1 == utils::intdiv_ceil(1, 3), "");
+static_assert(1 == utils::intdiv_ceil(3, 3), "");
+static_assert(2 == utils::intdiv_ceil(4, 3), "");
+static_assert(2 == utils::intdiv_ceil(4, 3), "");
+static_assert(0 == utils::intdiv_ceil(-1, 3), "");
+static_assert(-1 == utils::intdiv_ceil(-3, 3), "");
+static_assert(-1 == utils::intdiv_ceil(-4, 3), "");
+static_assert(2 == utils::intdiv_ceil(-4, -3), "");
+static_assert(2 == utils::intdiv_ceil(-5, -3), "");
+static_assert(0 == utils::intdiv_ceil(1, -3), "");
+static_assert(-1 == utils::intdiv_ceil(5, -3), "");
```

Comment:
you could add some more tests:
 * (0, negative number)
 * evenly divisible but not equal, eg. (6, 2) and variations: (-6, 2), (-6, -2), (6, -2)
 * test that division by 0 doesn't compile

Reply:
Not sure how can I test that 0 doesn't compile. Will do the rest.

Reply:
done in 585e874

Reply:
you could SFINAE it:

```
template<int n, int d, typename = void>
struct does_compile : std::false_type {};

template<int n, int d>
struct does_compile<n, d, decltype(intdiv_ceil(n, d))> : std::true_type {};
```

Reply:
```

template<int n, int d, typename = void>
struct does_compile : std::false_type {};

template<int n, int d>
struct does_compile<n, d, typename std::enable_if<(intdiv_ceil(n, d), true)>::type> : std::true_type {};
```

Reply:
I hope you don't mind me setting you as the author as you basically submitted the whole code in b11f101

Reply:
I'll survive somehow :) a slight addition would be to include a case that should compile so we could check for false negatives, e.g. if by accident `does_compile<...>::value` always returns `false`, basically a test for the test

Reply:
done in c26cf15